department-of-veterans-affairs/va.gov-cms
# VACMS-8499 fontawesome support

## Description

- Closes #8499. 
- Adds fontawesome support to the design system (and by extension, drupal).
- Added all the free icons from the set here: https://www.figma.com/file/ScABI31f3MRFEvQ8WitVPa/CMS-Design-System?node-id=925%3A45729 Some icons in that figma link, namely the outlined icons (info circle, warning trash can) are not available in the free library, as far as i can tell. 
- Fixes the tokens pages in storybook to update properly. Needed to split these to individual .stories.js files, otherwise twing wanted to use the last-loaded template for all stories in the file for reasons unknown.
- Cleans up the tokens display pages in storybook. As part of this cleanup, I moved some variables from the theme to the design system.
- Adds icons to design system alert component, this works in the cms as well.

**TODO: I will write up top-level docs for theme + design system + fontawesome for the repo soon. Things are scattered through a few readmes closer to the files at the moment.**

## Testing done
- Verified that bundle only includes the small subset of icons, not the full fontawesome library (3000+ free icons)
- Verified that fontawesome icons work in storybook and Drupal
- Verified that all stories hot reload as expected during local development
- Verified that styles and fonts still render & CSS properties override as expected in the CMS with some variables moved to design system

## Screenshots
<img width="594" alt="Screen Shot 2022-04-05 at 2 33 12 PM" src="https://user-images.githubusercontent.com/11279744/161858803-8424b8b3-ad50-4ada-b20c-e0463a338d09.png">

<img width="387" alt="Screen Shot 2022-04-05 at 2 32 45 PM" src="https://user-images.githubusercontent.com/11279744/161858794-a67564ae-d5b8-4c6f-813e-ab0696069c4d.png">
<img width="389" alt="Screen Shot 2022-04-05 at 2 32 49 PM" src="https://user-images.githubusercontent.com/11279744/161858798-a08908e5-2d17-44c4-818d-922d36d0c2fb.png">
<img width="371" alt="Screen Shot 2022-04-05 at 2 32 53 PM" src="https://user-images.githubusercontent.com/11279744/161858801-df6bf1f6-31b9-4091-920f-03661817737d.png">


## QA steps
- Visit the storybook link generated below, see that entries under "Design System Tokens" look much better than plain text lists.
- See that the alert component in storybook has appropriate icons for each state
- Visit drupal, see that the new icons show up in alerts in drupal. (Not the node link report, not updated yet)

### Definition of Done


### Select Team for PR review
